test/e2e/crd_admission_test.go

```diff
@@ -0,0 +1,129 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"strings"
+	"testing"
+
+	apiextensionsv1 "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"github.com/openshift/api/features"
+)
+
+// Test_GatewayAPICRDAdmissionOutsideCluster verifies that the ingress operator's ValidatingAdmissionPolicy
+// fails admission requests to modify Gateway API CRDs for a user which resides outside the cluster.
+func Test_GatewayAPICRDAdmissionOutsideCluster(t *testing.T) {
+	if gwapiEnabled, err := isFeatureGateEnabled(features.FeatureGateGatewayAPI); err != nil {
+		t.Fatalf("Failed to get Gateway API feature gate: %v", err)
+	} else if !gwapiEnabled {
+		t.Skipf("Skipping Test_GatewayAPICRDAdmissionOutsideCluster when GatewayAPI featuregate is not enabled")
+	}
+
+	testCRDs := []*apiextensionsv1.CustomResourceDefinition{
+		// managed CRDs
+		makeGWAPICRD("gatewayclass", "GatewayClass"),
+		makeGWAPICRD("gateway", "Gateway"),
+		makeGWAPICRD("httproute", "HTTPRoute"),
+		makeGWAPICRD("referencegrant", "ReferenceGrant"),
+		// unmanaged CRDs
+		//makeGWAPICRD("tcproute", "TCPRoute"),
```

Comment:
Technically we "manage" all CRDs in the group, at least to the effect that we don't allow anyone else to manage them.

Was the intention here to eventually test these other APIs? Or was it intentional to leave it as a comment to be illustrative for future code readers?

Reply:
> Was the intention here to eventually test these other APIs?

Yes, that was the idea. Technically it's a little more complicated because I will have to disable the cluster version operator to be able to create some "unmanaged" CRDs. Also, this will not test any new "code path": the `.spec.group` check is the same as for the "managed" CRDs. So I'm not fully convinced it's worth the effort.